src/Compilers/Core/Portable/InternalUtilities/StackGuard.cs

```diff
@@ -29,5 +30,27 @@ public static void EnsureSufficientExecutionStack(int recursionDepth)
                 RuntimeHelpers.EnsureSufficientExecutionStack();
             }
         }
+
+        internal static bool TryEnsureSufficientExecutionStack(int recursionDepth, bool throwOnFailure)
```

Comment:
>bool throwOnFailure [](start = 83, length = 19)

As a general principle, I'm not sure I think that this should be a parameter to this method. It seems like if you pass true here, you should have just checked it in the calling context.

Reply:
I don't think I feel strongly enough to hold up the PR, but it does make me a little leary based on uses.

---
In reply to: [358471427](https://github.com/dotnet/roslyn/pull/40367#discussion_r358471427) [](ancestors = 358471427)

Reply:
This pattern is a little strange but the purpose of passing `throwOnFailure` as a parameter rather than checking in the caller is to reduce the size of the caller's stack frame.

---
In reply to: [358471831](https://github.com/dotnet/roslyn/pull/40367#discussion_r358471831) [](ancestors = 358471831,358471427)

Reply:
> the purpose of passing throwOnFailure as a parameter rather than checking in the caller is to reduce the size of the caller's stack frame

That's important knowledge. Consider leaving a comment.

---
In reply to: [358914060](https://github.com/dotnet/roslyn/pull/40367#discussion_r358914060) [](ancestors = 358914060,358471831,358471427)